src/session/session_api.c

```diff
@@ -674,6 +671,77 @@ __session_alter(WT_SESSION *wt_session, const char *uri, const char *config)
     API_END_RET_NOTFOUND_MAP(session, ret);
 }
 
+/*
+ * __wt_session_blocking_checkpoint --
+ *     Perform a checkpoint or wait if it is already running to resolve an EBUSY error.
+ */
+int
+__wt_session_blocking_checkpoint(WT_SESSION_IMPL *session, bool force, uint64_t seconds)
+{
+    WT_DECL_RET;
+    WT_TXN_GLOBAL *txn_global;
+    uint64_t txn_gen;
+    const char *cfg[3] = {NULL, NULL, NULL};
+
+    cfg[0] = WT_CONFIG_BASE(session, WT_SESSION_checkpoint);
+    if (force)
+        cfg[1] = "force=1";
+
+    if ((ret = __wt_txn_checkpoint(session, cfg, false)) == 0)
+        return (0);
+    WT_RET_BUSY_OK(ret);
+
+    /*
+     * If there's a checkpoint running, wait for it to complete. If there's no checkpoint running or
+     * the checkpoint generation number changes, the checkpoint blocking us has completed.
+     */
+    txn_global = &S2C(session)->txn_global;
+    for (txn_gen = __wt_gen(session, WT_GEN_CHECKPOINT);; __wt_sleep(2, 0)) {
+        /*
+         * This loop only checks objects that are declared volatile, therefore no barriers are
+         * needed.
+         */
+        if (!txn_global->checkpoint_running || txn_gen != __wt_gen(session, WT_GEN_CHECKPOINT))
+            break;
+
+        /* If there's a timeout, give up. */
+        if (seconds == 0)
+            continue;
+        seconds -= 2;
+        if (seconds <= 2)
+            return (EBUSY);
```

Comment:
Seconds is unsigned, so I think this can turn into an infinite loop if it's called with `seconds == 1`. Maybe:
```
if (seconds == 0)
    continue;
if (seconds <= 2)
    return (EBUSY);
seconds -= 2;
```

Reply:
Also, it seems like this could be a value we'd want to change sometime and the uses are not all nearby. Maybe use a macro `#define WT_CKPT_WAIT 2`.

Reply:
Thanks. Fixed and used the macro.